app/src/main/java/org/projectbuendia/client/ui/chart/ChartRenderer.java

```diff
@@ -191,18 +210,71 @@ void addOrders(List<Order> orders) {
 
         /** Returns the column that contains the given instant, creating it if it doesn't exist. */
         Column getColumnContainingTime(ReadableInstant instant) {
-            LocalDate date = new DateTime(instant).toLocalDate();  // a day in the local time zone
-            DateTime start = date.toDateTimeAtStartOfDay();
+            Column returnColumn;
+            switch (mChartColumnType) {
+                case Column.ENCOUNTER:
+                    returnColumn =  getEncounterColumn(instant);
+                    break;
+                case Column.TIMED:
+                    returnColumn =  getTimedColumn(instant);
+                    break;
+                case Column.DAILY:
+                default:
+                    returnColumn =  getDailyColumn(instant);
+                    break;
+            }
+            return returnColumn;
+        }
+
+        Column getDailyColumn(ReadableInstant instant) {
+            LocalDateTime date = new DateTime(instant).toLocalDateTime();  // a day in the local time zone
+            DateTime start = date.withTime(0, 0, 0, 0).toDateTime();
```

Comment:
I think you can use `LocalDateTime.withTimeAtStartOfDay` instead of `withTime(0,0,0,0)`